tls/s2n_record_write.c

```diff
@@ -526,3 +524,11 @@ int s2n_record_write(struct s2n_connection *conn, uint8_t content_type, struct s
     iov.iov_len = in->size;
     return s2n_record_writev(conn, content_type, &iov, 1, 0, in->size);
 }
+
+S2N_RESULT s2n_record_write_all(struct s2n_connection *conn, uint8_t content_type, struct s2n_blob *in)
```

Comment:
Why did you call this function s2n_record_write_all? Write all what?

Reply:
s2n_record_write_all enforces that all data / the full blob is written to the record. s2n_record_write only writes as much data as it can fit in a single record, potentially not writing the full blob. Any ideas for a clearer name?

Reply:
I guess my next question is, why is this function necessary?

Reply:
Addressed by https://github.com/aws/s2n-tls/pull/3708#discussion_r1051930519